### ISSUE TITLE
support use-rp option in vic-machine create/inspect/configure/upgrade

### ISSUE THREAD

**Issue by emlin** [vmware/vic#6272]
@hickeng as your request, we'll need to support --use-rp in create/configure/delete. We already support it in create/delete with one recent bug fix #6250 

But the problem is that how to support it in configure command.
I still didn't find one way to convert vApp to Resource Pool or vise versa. So if we want to update VCH configuration from using vApp to Resourse Pool, we'll need to create new resource pool, and move VCH together with containers to the new resource pool.

Feels like it's possible, but what if anything error happens in the middle, I didn't see good protection way for that level's operation.
So I'm wondering what's the use case to solve through this reconfiguration.

Other issues needs to be solved related to --use-rp
- [ ] if create with --use-rp and --force, vic-machine will ignore resource pool creation failure, and continue to deploy VCH in parent resource pool.
The problem is that this VCH is not recognized by vic-machine anymore, cause the resource pool convention is broken. To fix this problem, we should fail deployment even --force is specified.
- [ ] add CI test for --use-rp in create/upgrade/delete, to avoid regression

**Comment by emlin** [comment]
@mdubya66 upgrade works if vch is created in resource pool, it should be tested as well to avoid regression

**Comment by hickeng** [comment]
@emlin I mean we need to support a VCH created using --use-rp in those commands, not support adding the command to those commands.
We had to use `--id=` option in configure/inspect/delete after deploying into a resource pool, the `--name= --compute-resource=` pair didn't work.

**Comment by emlin** [comment]
talked with @hickeng, this is caused by resource pool creation failure, after fix of #6250, the other commands all work well.

to avoid this confusion, should make following change to return error if resource pool creation failed.
```
diff --git a/lib/install/management/create.go b/lib/install/management/create.go
index 52a9c83..062150e 100644
--- a/lib/install/management/create.go
+++ b/lib/install/management/create.go
@@ -87,14 +87,7 @@ func (d *Dispatcher) createPool(conf *config.VirtualContainerHostConfigSpec, set
        } else {
                if d.vchPool, err = d.createResourcePool(conf, settings); err != nil {
                        detail := fmt.Sprintf("Creating resource pool failed: %s", err)
-                       if !d.force {
-                               return errors.New(detail)
-                       }
-
-                       log.Error(detail)
-                       log.Errorf("Deploying vch under parent pool %q, (--force=true)", settings.ResourcePoolPath)
-                       d.vchPool = d.session.Pool
-                       conf.ComputeResources = append(conf.ComputeResources, d.vchPool.Reference())
+                       return errors.New(detail)
                }
        }
```

**Comment by mhagen-vmware** [comment]
The beginnings of a test for this are already here:
https://github.com/vmware/vic/pull/6237
